### PR TITLE
feat(listviews): SIP-34 filters for charts, dashboards, datasets

### PR DESCRIPTION
**Pull request by nytai**
### SUMMARY

- api endpoint `/api/v1/database/schemas` for fetching list of schemas.
- adds SIP-34 filters to datasets list view
- removes `LIST_VIEWS_SIP34_FILTER_UI` feature flag
- enabled SIP-34 filters for datasts, charts, dashboards, listview
- moves `src/views/*` -> `src/views/CRUD/*` (eg, `src/views/datasetList/DatasetList.tsx` -> `src/views/CRUD/dataset/DatasetList.tsx` ) -- idea being that `src/views/CRUD` namespace can share code, ie, data fetching logic, state hooks, edit modals, delete modals, etc
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Screen Shot 2020-07-15 at 6 46 41 PM
Screen Shot 2020-07-15 at 6 46 49 PM

### TEST PLAN

- unit tests pass
- can filter on charts, dashboards, datasets list views
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI: Behind `ENABLE_REACT_CRUD_VIEWS` flag
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by bkyryliuk** [comment]
can't wait to see it live!

File: superset/databases/api.py [hunk 1]
```diff
@@ -265,3 +275,87 @@ def select_star(
...
+              application/json:
+                schema:
+                  type: object
+                  properties:
+                    page_size:
+                      type: integer
+                    page:
+                      type: integer
+                    filter:
+                      type: string
```
Comment (dpgaspar) [inline superset/databases/api.py:301]: It's nicer to use a reference here to the schema, so we don't have to repeat ourselfs. Take a look at: https://github.com/apache/incubator-superset/blob/master/superset/charts/api.py#L502

File: superset/databases/api.py [hunk 2]
```diff
@@ -265,3 +275,87 @@ def select_star(
...
+                    properties:
+                      count:
+                        type: integer
+                      result:
+                        type: object
+                        properties:
+                          value:
+                            type: string
+                          text:
+                            type: string
```
Comment (dpgaspar) [inline superset/databases/api.py:318]: Here also, declare a marshmallow schema for the response and reference that. Register your schema on OpenAPI using `openapi_spec_component_schemas` FAB attr

**Review by dpgaspar — COMMENTED** [review]
first pass, couple of comments

**Comment by mistercrunch** [comment]
This looks amazing, let's get it through! Looks like you're one cypress test away :)

**Review by rusackas — APPROVED** [review]
A few little nits/questions, but in general this LGTM! Excited to see these changes!!

File: superset-frontend/src/views/CRUD/dataset/DatasetList.tsx [hunk 3]
```diff
@@ -69,128 +66,191 @@ interface DatasetListProps {
...
+    });
+
+    return json?.result?.map(
+      ({ database_name: label, id: value }: Database) => ({
+        label,
+        value,
+      }),
+    );
+  } catch (e) {
+    console.error(e);
```
[the remark this answers is not in the document]
Comment (nytai) [reply superset-frontend/src/views/CRUD/dataset/DatasetList.tsx:114]: yea I should probably move to using `handleError` in these views. Will look into a followup PR for doing that as it'll require some testing.

File: superset-frontend/src/views/CRUD/utils.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * under the License.
+ */
+import { SupersetClient } from '@superset-ui/connection';
+import rison from 'rison';
+
+export const createFetchOwners = (
+  resource: string,
+  handleError: (error: Response) => void,
+) => async (filterValue = '', pageIndex?: number, pageSize?: number) => {
+  const resourceEndpoint = `/api/v1/${resource}/related/owners`;
```
[the remark this answers is not in the document]
Comment (nytai) [reply superset-frontend/src/views/CRUD/utils.tsx:26]: yup, will look into doing that. Hoping to have some common functions/hooks shared between the various listviews related to error handling, data fetching, permission checking, etc.

**Comment by rusackas** [comment]
Impacts #8976

**Pull request merged**